src/input.service.ts

```diff
@@ -289,76 +289,66 @@ export class InputService {
-      138, // NavigationUp
-    ]],
-    [Direction.DOWN, [
-      40,  // UpArrow
-      212, // GamepadLeftThumbstickDown
-      204, // GamepadDPadDown
-      139, // NavigationDown
-    ]],
-    [Direction.SUBMIT, [
-      13,  // Enter
-      32,  // Space
-      142, // NavigationAccept
-      195, // GamepadA
-    ]],
-    [Direction.BACK, [
-      8,   // Backspace
-      196, // GamepadB
-    ]],
-    [Direction.X, [
-      103, // Numpad 7
-      197, // GamepadX
-    ]],
-    [Direction.Y, [
-      105,   // Numpad 9
-      198, // GamepadY
-    ]],
-    [Direction.TABLEFT, [
-      100, // Numbpad Left
-      200, // Left Bumper
-    ]],
-    [Direction.TABRIGHT, [
-      102, // Numpad Right
-      199, // Right Bumper
-    ]],
-    [Direction.TABUP, [
-      104, // Numpad Up
-      201, // Left Trigger
-    ]],
-    [Direction.TABDOWN, [
-      98, // Numpad Down
-      202, // Right Trigger
-    ]],
-    [Direction.VIEW, [
-      111, // Numpad Divide
-      208, // View Button
-    ]],
-    [Direction.MENU, [
-      106, // Numpad Multiply
-      207, // Menu Button
-    ]],
-  ]);
+  public directionFromCode(keyCode: number): Direction {
+    switch (keyCode) {
+      case 37:  // LeftArrow
+      case 214: // GamepadLeftThumbstickLeft
+      case 205: // GamepadDPadLeft
+      case 140: // NavigationLeft
+        return Direction.LEFT;
+      case 39:  // RightArrow
+      case 213: // GamepadLeftThumbstickRight
+      case 206: // GamepadDPadRight
+      case 141: // NavigationRight
+        return Direction.RIGHT;
+      case 38:  // UpArrow
+      case 211: // GamepadLeftThumbstickUp
+      case 203: // GamepadDPadUp
+      case 138: // NavigationUp
+        return Direction.UP;
+      case 40:  // DownArrow
+      case 212: // GamepadLeftThumbstickDown
+      case 204: // GamepadDPadDown
+      case 139: // NavigationDown
+        return Direction.DOWN;
+      case 13:  // Enter
+      case 32:  // Space
+      case 142: // NavigationAccept
+      case 195: // GamepadA
+        return Direction.SUBMIT;
+      case 8:   // Backspace
+      case 196: // GamepadB
+        return Direction.BACK;
+      case 103: // Numpad 7
+      case 197: // GamepadX
+        return Direction.X;
+      case 105: // Numpad 9
+      case 198: // GamepadY
+        return Direction.Y;
+      case 100: // Numbpad Left
+      case 200: // Left Bumper
+        return Direction.TABLEFT;
+      case 102: // Numpad Right
+      case 199: // Right Bumper
+        return Direction.TABRIGHT;
+      case 104: // Numpad Up
+      case 201: // Left Trigger
+        return Direction.TABUP;
+      case 98:  // Numpad Down
+      case 202: // Right Trigger
+        return Direction.TABDOWN;
+      case 111: // Numpad Divide
+      case 208: // View Button
+        return Direction.VIEW;
+      case 106: // Numpad Multiply
+      case 207: // Menu Button
+        return Direction.MENU;
+      default:
+        return null;
+    }
+  }
```

Comment:
The reason why I kept that list of EventEmitters is to avoid a breaking change in API, but I guess it is better to keep it as a map. I'll need to PR frontend2 as well then